### PR TITLE
Feature/health checks and signals

### PR DESCRIPTION
**Pull request by davent**

### CODE REVIEW

**Comment by kriswehner** [comment]
@davent Can we get a test on this? I'd love to merge it, but would like it to have test coverage.

**Comment by davent** [comment]
Sure thing!

**Comment by davent** [comment]
Have added some tests to the main server test suite.

File: changeserver/server_main_test.go [hunk 1]
```diff
@@ -169,3 +176,32 @@ const testAlternateScopeTableSQL = `
 
 const dropTableSQL = "drop table changeserver_test"
 const dropAlternativeScopeTableSQL = "drop table changeserver_scope_test"
+
+func testHealthChecks() {
+	// The server should start in a healthy state
+	Expect(testServer.checkHealth()).To(Equal(goscaffold.OK), "The server should be marked as healthy")
+
+	// Send the process a SIGUSR1 to mark it as unhealthy
+	syscall.Kill(syscall.Getpid(), syscall.SIGUSR1)
```
Comment (chuckg) [inline changeserver/server_main_test.go]: I think syscall stuff is deprecated'ish as of 1.4: 
https://golang.org/pkg/syscall/

The os package has a platform agnostic version as well:
https://golang.org/pkg/os/#Getpid

**Comment by davent** [comment]
I am not sure it is "deprecated" per se. They use the term "This package is locked down". However, the os package itself uses the syscall package both for the Signal type and also in it's own tests: https://golang.org/src/os/signal/signal_test.go (see line 41 for an example)

File: changeserver/server_main_test.go [hunk 2]
```diff
@@ -169,3 +176,32 @@ const testAlternateScopeTableSQL = `
 
 const dropTableSQL = "drop table changeserver_test"
 const dropAlternativeScopeTableSQL = "drop table changeserver_scope_test"
+
+func testHealthChecks() {
```
Comment (davent) [inline changeserver/server_main_test.go]: NP. Done

**Comment by kriswehner** [comment]
👍

**Pull request merged**